packages/controller/src/resources/monitoring/tenants/grafanaDatasources.ts

```diff
@@ -269,44 +242,52 @@ http {
             location / {
               ${generalProxyConfig}
 
-              resolver ${dnsResolver};
-              set $backend http://querier.loki.svc.cluster.local:1080;
-              proxy_pass $backend;
+              proxy_pass http://querier.loki.svc.cluster.local:1080;
             }
 
             # For specific routes, pass on to the ruler
             location /prometheus/api/v1/rules {
               ${generalProxyConfig}
-              resolver ${dnsResolver};
               # Do not change the request URL
-              set $backend http://ruler.loki.svc.cluster.local:1080;
-              proxy_pass $backend;
+              proxy_pass http://ruler.loki.svc.cluster.local:1080/prometheus/api/v1/rules;
+            }
+
+            location /prometheus/api/v1/rules/ {
+              ${generalProxyConfig}
+              # Do not change the request URL
+              proxy_pass http://ruler.loki.svc.cluster.local:1080/prometheus/api/v1/rules/;
             }
 
             location /prometheus/api/v1/alerts {
               ${generalProxyConfig}
-              resolver ${dnsResolver};
               # Do not change the request URL
-              set $backend http://ruler.loki.svc.cluster.local:1080;
-              proxy_pass $backend;
+              proxy_pass http://ruler.loki.svc.cluster.local:1080/prometheus/api/v1/alerts;
+            }
+
+
+            location /prometheus/api/v1/alerts/ {
+              ${generalProxyConfig}
+              # Do not change the request URL
+              proxy_pass http://ruler.loki.svc.cluster.local:1080/prometheus/api/v1/alerts/;
             }
 
             location /loki/api/v1/rules {
               ${generalProxyConfig}
-              resolver ${dnsResolver};
-              # Do not change the request URL, this works for the following endpoints:
-              # /loki/api/v1/rules
-              # /loki/api/v1/rules/{namespace}
-              set $backend http://ruler.loki.svc.cluster.local:1080;
-              proxy_pass $backend;
+
+              proxy_pass http://ruler.loki.svc.cluster.local:1080/loki/api/v1/rules;
+            }
+
+            location /loki/api/v1/rules/ {
+              ${generalProxyConfig}
+
+              proxy_pass http://ruler.loki.svc.cluster.local:1080/loki/api/v1/rules/;
             }
 
             # Loki legacy route for saving rules (legacy routes which the Grafana 8 alerting feature uses)
-            location ~ ^/api/prom/rules(?<urlsuffix>.*)$ {
```

Comment:
Using a variable in the proxy_pass requires us to set the DNS resolver otherwise nginx returns a 502 and logs the error no resolver defined to resolve. This is why I reverted back to setting the location explicitly here and in the other instances.